pandas/core/dtypes/concat.py

```diff
@@ -132,6 +183,73 @@ def is_nonempty(x) -> bool:
     return np.concatenate(to_concat, axis=axis)
 
 
+def concat_arrays(to_concat: List[Any]) -> ArrayLike:
```

Comment:
List[Any] -> List[ArrayLike]?

Reply:
With the `NullArrayProxy` handling, this is quite annoying to add more specific types